src/io/openshift/Utils.groovy

```diff
@@ -19,7 +19,7 @@ class Utils {
     def resources = [resource].flatten()
 
     resources.each { r ->
-      def kind = r && r.kind ? r.kind.toLowerCase() : "test"
+      def kind = r && r.kind ? r.kind.toLowerCase() : ""
```

Comment:
why not call it "unknown" ? 
shouldn't there be a kind for all resources? isn't that mandatory?

Reply:
It's mandatory.
In current case if `kind` is empty then pipeline terminates with "NullPointerException" and its hard to understand why its failed.
With this patch pipeline fails with appropriate error message.

As you have suggested it better to handle the validation in`processTemplate` and `loadResoucres` API's

Reply:
I would rather that we have a `validate()` method that asserts presence of `kind` and fail before applying the resource if the field is mandatory than do a hack like this.

Reply:
will fix this